python-checks/src/main/java/org/sonar/python/checks/cdk/PublicApiIsSecuritySensitiveCheck.java

```diff
@@ -19,30 +19,97 @@
+  private static final String AUTHORIZATION_TYPE = "authorization_type";
+  private static final String AUTHORIZATION_TYPE_NONE = "aws_cdk.aws_apigateway.AuthorizationType.NONE";
+
+  private Set<String> safeMethods = new HashSet<>();
 
   @Override
   protected void registerFqnConsumer() {
     checkFqns(List.of("aws_cdk.aws_apigateway.CfnMethod", "aws_cdk.aws_apigatewayv2.CfnRoute"), (subscriptionContext, callExpression) ->
-      getArgument(subscriptionContext, callExpression, "authorization_type").ifPresentOrElse(
+      getArgument(subscriptionContext, callExpression, AUTHORIZATION_TYPE).ifPresentOrElse(
         argument -> argument.addIssueIf(isString("NONE"), MESSAGE),
         () -> subscriptionContext.addIssue(callExpression.callee(), OMITTING_MESSAGE)
       )
     );
+    checkFqns(List.of("aws_cdk.aws_apigateway.RestApi", "aws_cdk.aws_apigateway.Resource.add_resource"), (subscriptionContext, callExpression) ->
+      getArgument(subscriptionContext, callExpression, "default_method_options").ifPresent(
+        argument -> {
+          if (isSafe(subscriptionContext, argument)) {
+            // if invocation of RestApi() or Resource.add_resource() is with the safe default, and it's in the method,
+            // then store the method's full qualified name in safeMethods
+            enclosingMethodFqn(callExpression).ifPresent(fqn -> safeMethods.add(fqn));
+          }
+        }
+      )
+    );
     checkFqn("aws_cdk.aws_apigateway.Resource.add_method", (subscriptionContext, callExpression) ->
-      getArgument(subscriptionContext, callExpression, "authorization_type").ifPresent(
-        argument -> argument.addIssueIf(isFqn("aws_cdk.aws_apigateway.AuthorizationType.NONE"), MESSAGE)
+      getArgument(subscriptionContext, callExpression, AUTHORIZATION_TYPE).ifPresentOrElse(
+        argument -> argument.addIssueIf(isFqn(AUTHORIZATION_TYPE_NONE), MESSAGE),
+        () -> enclosingMethodFqn(callExpression).filter(fqn -> safeMethods.contains(fqn)).ifPresentOrElse(fqn -> {}, () ->
+          // if the invocation of Resource.add_method() is without explicit authorization_type argument,
+          // and it's not in the scope of safeMethod the call, then it's considered unsafe
+          // this is the best possible approximation for now
+          subscriptionContext.addIssue(callExpression.callee(), OMITTING_MESSAGE)
+        )
       )
     );
   }
+
+  private static Optional<String> enclosingMethodFqn(Tree tree) {
+    FunctionDef functionDef = (FunctionDef) TreeUtils.firstAncestorOfKind(tree, Tree.Kind.FUNCDEF);
+    return Optional.ofNullable(functionDef)
+      .map(FunctionDefImpl.class::cast)
+      .map(FunctionDefImpl::functionSymbol)
+      .map(Symbol::fullyQualifiedName);
+  }
+
+  private static boolean isSafe(SubscriptionContext subscriptionContext, CdkUtils.ExpressionFlow argument) {
```

Comment:
Maybe add an entity name about what exactly is safe. You store the fqn in a field called `safeMethods` maybe `isSafeMethod` is a proper name?